### PR TITLE
Proposing CredentialProvider interface

### PR DESCRIPTION
**Pull request by kunkun-tang**
Creating Interface in order to let custom credential provider register user's secret keys. 

One sample use case: Today we don't have a secure way to pass on credentials into Hadoop. This interface allows Azkaban to fetch user's credential from external system (like certificate authority), and add it into Hadoop Job Context. Then submitted hadoop job will be able to use it.

Today we only have one method, but could have more in future.

This PR only proposes an interface, and the next step will be to use this interface in `HadoopSecurityManager_H_2_0` class.

Note: the changes in HadoopSecurityManager_H_2_0.java are automatically made by `save action` IntelliJ plugin.

### CODE REVIEW

File: az-core/src/main/java/azkaban/Constants.java [hunk 1]
```diff
@@ -161,6 +161,8 @@
 
     // enable Quartz Scheduler if true.
     public static final String ENABLE_QUARTZ= "azkaban.server.schedule.enable_quartz";
+
+    public static final String CUSTOM_CREDENTIAL_NAME = "azkaban.security.credential";
```
Comment (jamiesjc) [inline az-core/src/main/java/azkaban/Constants.java]: Where would this be configured? In azkaban.properties?

Comment (kunkun-tang) [reply az-core/src/main/java/azkaban/Constants.java]: No, this will be configured under jobtype/commonprivate.properties.

**Comment by jamiesjc** [comment]
What is the custom credential provider? Can you give a little more detail on what you are planning to do with this new interface?

**Comment by kunkun-tang** [comment]
Description Updated. Thanks. @jamiesjc

File: azkaban-hadoop-security-plugin/src/main/java/azkaban/security/CredentialProvider.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+package azkaban.security;
+
+/**
+ * Fetch and inject user secret key into custom credential object.
+ */
+public interface CredentialProvider {
+
+  /**
+   * register custom secret keys on behalf of an user into Credentials.
```
Comment (chengren311) [inline azkaban-hadoop-security-plugin/src/main/java/azkaban/security/CredentialProvider.java]: lowercase credentials

[the remark this answers is not in the document]
Comment (kunkun-tang) [reply azkaban-hadoop-security-plugin/src/main/java/azkaban/security/CredentialProvider.java]: corrected!

**Pull request merged**